client/my-sites/stats/activity-log-item/activity-icon.jsx

```diff
@@ -21,23 +22,28 @@ export default class ActivityIcon extends PureComponent {
 			'widget',
 		] ).isRequired,
 		name: PropTypes.string.isRequired,
+		object: PropTypes.object.isRequired,
 	};
 
 	getIcon() {
 		const {
 			group,
 			name,
+			object,
 		} = this.props;
 
 		switch ( name ) {
 			// Inline return makes alphabetizing and searching easier :)
 			case 'post__published': return 'create';
 			case 'post__trashed': return 'trash';
-			case 'user__registered': return 'user-add';
 		}
 
 		switch ( group ) {
 			case 'attachment':
+				if ( 'application/zip' === get( object, [ 'attachment', 'mime_type' ] ) ) {
+					return 'attachment';
```

Comment:
It may be simpler and more robust to identify images rather what's not an image.

```js
if ( startsWith( get( object, [ 'attachment', 'mime_type' ] ), 'image/' ) ) {
  return 'image';
}
return 'attachment';
```